File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeNodes(self, head):
        x = ListNode(0)
        curr = x
        tmp = 0

        while head:
            if head.val == 0:
                if tmp != 0:
                    curr.next = ListNode(tmp)
                    curr = curr.next
                    tmp = 0
            else:
                tmp += head.val
            head = head.next
        
        return x.next

File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* mergeNodes(struct ListNode* head) {
    struct ListNode* x = (struct ListNode*)malloc(sizeof(struct ListNode));
    x->val = 0;
    x->next = NULL;
    struct ListNode* curr = x;
    int tmp = 0;

    while (head != 0) {
        if (head->val == 0) {
            if (tmp != 0) {
                curr->next = (struct ListNode*)malloc(sizeof(struct ListNode));
                curr->next->val = tmp;
                curr->next->next = NULL;
                curr = curr->next;
                tmp = 0;
            }
        } else {
            tmp += head->val;
        }
        head = head->next;
    }

    return x->next;
}